src/clustering/pqkmeans.h

```diff
@@ -45,6 +45,7 @@ class PQKMeans {
     float L2SquaredDistance(const std::vector<float> &vec1,
                             const std::vector<float> &vec2);
 
+    void SetClusterCenters(std::vector<std::vector<unsigned char>> centers_new);
```

Comment:
and change this line to a const-reference, too:   `const std::vector::vector<unsigned char>> &centers_new`